File: LeetCode/python/_461. Hamming Distance.py
#第一想法是都转为2进制字符串了比较，但想想应该有更快的，比如什么算差之类的
#好吧，查了个一行的。，还是转2进制，不过利用python 的库，bin()转后还是字符串的格式
class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        return bin(x^y).count('1')

File: LeetCode/C_plus/_461.HammingDistance.cpp
# 3 ms -24%
# c++中可直接对十进制数字进行位运算
class Solution {
public:
    int hammingDistance(int x, int y) {
       int z=x^y;
        //^符号表示：按位异或运算符。
       int count=0;
       while(z != 0 ) {
         if (z&1)count++;
         z>>=1;
       }  
       return count; 
    }
};
